python/ray/data/dataset.py

```diff
@@ -1694,8 +1695,16 @@ def write_datasource(self, datasource: Datasource[T],
             write_args: Additional write args to pass to the datasource.
         """
 
+        ctx = DatasetContext.get_current()
         blocks, metadata = zip(*self._blocks.get_blocks_with_metadata())
-        write_results = datasource.do_write(blocks, metadata, **write_args)
+
+        # Prepare write in a remote task so that in Ray client mode, we aren't
```

Comment:
What's the cost for non-ray client mode?

Reply:
Shouldn't matter much (a few ms)